main/db.c

```diff
@@ -408,11 +411,13 @@ static int db_get_common(const char *family, const char *key, char **buffer, int
 		res = -1;
 	} else {
 		const char *value = (const char *) result;
-
-		if (bufferlen == -1) {
-			*buffer = ast_strdup(value);
-		} else {
-			ast_copy_string(*buffer, value, bufferlen);
+		/* If buffer is null, we're just testing for existence */
+		if (buffer != NULL) {
+			if (bufferlen == -1) {
+				*buffer = ast_strdup(value);
+			} else {
+				ast_copy_string(*buffer, value, bufferlen);
+			}
```

Comment:
This change seems unrelated? Perhaps from a previous revision?

Reply:
There was a reason for this but I don't remember what it was.  I'll investigate.